pynamodb/connection/base.py

```diff
@@ -1020,7 +1020,15 @@ def rate_limited_scan(self,
                             if not limit:
                                 return
 
-                    latest_scan_consumed_capacity = data.get(CONSUMED_CAPACITY).get(CAPACITY_UNITS)
+                    if CONSUMED_CAPACITY in data:
```

Comment:
I just want to call out that I think the most concerning problem with this is that if for whatever reason a real production use case ends up lacking this information (bug in client or server) we'll just happily truck on with effectively silent failure rather than explicitly failing.